dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -394,24 +396,18 @@ public Workflow setupWorkflowVersions(String repositoryId, Workflow workflow, Op
 
         // when getting a full workflow, look for versions and check each version for valid workflows
         List<Triple<String, Date, String>> references = new ArrayList<>();
-        try {
-            GHRef[] refs = repository.getRefs();
-            for (GHRef ref : refs) {
-                Triple<String, Date, String> referenceTriple = getRef(ref, repository);
-                if (referenceTriple != null) {
-                    if (versionName.isEmpty() || Objects.equals(versionName.get(), referenceTriple.getLeft())) {
-                        references.add(referenceTriple);
-                    }
+
+        GHRef[] refs = getBranchesAndTags(repository);
+        for (GHRef ref : refs) {
+            Triple<String, Date, String> referenceTriple = getRef(ref, repository);
+            if (referenceTriple != null) {
+                if (versionName.isEmpty() || Objects.equals(versionName.get(), referenceTriple.getLeft())) {
+                    references.add(referenceTriple);
                 }
             }
-        } catch (GHFileNotFoundException e) {
```

Comment:
See my comments about how I think `getBranchesAndTags` should throw an exception.

Reply:
Could you not leave this section as it was, with both catches?